src/include/duckdb/function/macro_function.hpp

```diff
@@ -20,6 +22,7 @@ class MacroCatalogEntry;
 class MacroFunction {
 public:
 	explicit MacroFunction(unique_ptr<ParsedExpression> expression);
```

Comment:
@Mytherin 
I think maybe have a misunderstanding here.
In this PR   in  **bind_table_function.cpp / Bind(TableFunctionRef &ref)** . Now reading a possible  TableMacro entry from the TableFunction catalog.  

If you follow the token TABLE_MACRO_ENTRY:
In  **schema_catalog_entry.cpp / GetCatalogSet()**  I am setting the Catalog to table_functions  for a TableMacro

....

Reply:
You are right that this is partially resolved - the first two bullet points no longer apply. The general point still stands that currently the code uses the `MacroFunction` as two different things (either a scalar macro function or a table macro function). The same is the case with the `MacroCatalogEntry`. I think both of those cases would be cleaner if they would instead be separate classes (`TableMacroCatalogEntry` and `ScalarMacroCatalogEntry`) that inherit from the same base class, as mentioned in the comment above.

Reply:
Would you be  prepared to stick with the existing code ? 
Refactoring with inheritance will involve creating two base classes, four derived classes, possibly a template version of CreateFunctionInfo; and then   edits, some big some small in the following files:

catalog_type.cpp 
duckdb_functions.cpp 
create_functions.cpp 
transform_drop.cpp
bind_create.cpp
bind_table_function.cpp
checkpoint_manager.cpp
wal_replay.cpp 
commit_state.cpp 

And probably two or three Pull Request to get it all right ?

Reply:
I don't think this refactor is particularly big - it is mostly extracting some functionality into a base class and cleaning up the code. Currently a lot of stuff is changed unnecessarily (e.g. the storage version, code relating to scalar macro's) which I would really prefer not to be touched. 

Why would you need to open up more pull requests? Could you please update the current one, instead of opening new pull requests for the same feature? This allows the comments to stay in one place and is much cleaner/simpler.